Multi-chain docs and FAQs (#2487)
dfinity/portal: docs/developer-docs/integrations/multi-chain/cketh-faq.md

The main difference between a ckETH transaction and a regular ETH transaction lies in the transaction fees and the blockchain they operate on. Regular ETH transactions occur on the Ethereum network and are subject to Ethereum's gas fees, which can be quite high. On the other hand, ckETH transactions occur on the Internet Computer blockchain, which has significantly lower transaction fees.

Yes, you can use ckETH just like you would use ETH, but within the Internet Computer ecosystem. You can deposit your ETH to a specific function in the ckETH helper contract on Ethereum, which then triggers the ICP ckETH canister smart contract to mint the same amount of ckETH to the specified ICP principal or wallet address. This ckETH can then be used natively on the Internet Computer, and can be converted back to ETH at any time.

However, it's important to note that while ckETH can be used in a similar manner to ETH, the platforms and services that accept ckETH might be different from those that accept ETH, as they are operating on different blockchains.

> : [A Data-Driven Exploration of ckETH](https://medium.com/dfinity/a-data-driven-exploration-of-ckETH-the-digital-twin-of-ether-on-the-internet-computer-36b762be72e7#b197)
> : [How does ckETH work?](https://medium.com/dfinity/a-data-driven-exploration-of-ckETH-the-digital-twin-of-ether-on-the-internet-computer-36b762be72e7#e978)
> : [How to Acquire ckETH](https://medium.com/dfinity/how-to-acquire-ckETH-02d863c835fc#e10f)
> : [Ethereum &lt;&gt; ICP > Multi-chain DeFi](https://internetcomputer.org/ethereum-integration)

Reviewer: ```suggestion
- [A Data-Driven Exploration of ckETH](https://medium.com/dfinity/a-data-driven-exploration-of-ckETH-the-digital-twin-of-ether-on-the-internet-computer-36b762be72e7#b197)
- [How does ckETH work?](https://medium.com/dfinity/a-data-driven-exploration-of-ckETH-the-digital-twin-of-ether-on-the-internet-computer-36b762be72e7#e978)
- [How to Acquire ckETH](https://medium.com/dfinity/how-to-acquire-ckETH-02d863c835fc#e10f)
- [Ethereum + ICP > Multi-chain DeFi](https://internetcomputer.org/ethereum-integration)
```

## What are the benefits of using ckETH over traditional Bitcoin transactions? 

> The benefits of using ckETH over traditional Bitcoin transactions primarily revolve around transaction speed, cost, and interoperability with the Internet Computer ecosystem.

Reviewer: ```suggestion
The benefits of using ckETH over traditional Bitcoin transactions primarily revolve around transaction speed, cost, and interoperability with the ICP ecosystem.
```

> 1. **Lower Transaction Fees**: One of the main advantages of using ckETH is the significantly lower transaction fees compared to Bitcoin. Bitcoin transactions can sometimes be expensive, especially during periods of high network congestion. On the other hand, ckETH transactions occur on the Internet Computer blockchain, which has significantly lower transaction fees.
> 
> 2. **Faster Transaction Speed**: The Internet Computer blockchain has a faster transaction speed compared to the Bitcoin network. This means that transactions using ckETH can be processed more quickly, providing a smoother and more efficient user experience.
> 
> 3. **Interoperability**: ckETH allows for seamless interaction between the Ethereum and Internet Computer ecosystems. This interoperability can potentially unlock new use cases and applications, such as decentralized finance (DeFi) platforms that span multiple blockchains.
> 
> 4. **Usability**: The process of acquiring and using ckETH is designed to be straightforward and user-friendly, removing many of the barriers that can deter people from engaging with blockchain technology.
> 
> However, it's important to note that while ckETH provides these benefits, it's not a direct replacement for Bitcoin or ETH. Each has its own unique features and use cases, and the choice between them will depend on your specific needs and circumstances.
> 
> : [A Data-Driven Exploration of ckETH](https://medium.com/dfinity/a-data-driven-exploration-of-ckETH-the-digital-twin-of-ether-on-the-internet-computer-36b762be72e7#b197)
> : [The Purposes of the Internet Computer](https://medium.com/dfinity/announcing-internet-computer-mainnet-and-a-20-year-roadmap-790e56cbe04a#0289)
> : [Internet Computer for Ethereum Developers](https://wiki.internetcomputer.org/wiki/The_Internet_Computer_for_Ethereum_Developers#A_very_brief_introduction_to_the_Internet_Computer)
> : [Extend Bitcoin, Ethereum and other blockchains](https://wiki.internetcomputer.org/wiki/Extend_Bitcoin,_Ethereum_and_other_blockchains#firstHeading)

Reviewer: ```suggestion
1. **Lower transaction fees**: One of the main advantages of using ckETH is the significantly lower transaction fees compared to Bitcoin. Bitcoin transactions can sometimes be expensive, especially during periods of high network congestion. On the other hand, ckETH transactions occur on the Internet Computer blockchain, which has significantly lower transaction fees.

2. **Faster transaction speed**: The ICP blockchain has a faster transaction speed compared to the Bitcoin network. This means that transactions using ckETH can be processed more quickly, providing a smoother and more efficient user experience.

3. **Interoperability**: ckETH allows for seamless interaction between the Ethereum and ICP ecosystems. This interoperability can potentially unlock new use cases and applications, such as decentralized finance (DeFi) platforms that span multiple blockchains.

4. **Usability**: The process of acquiring and using ckETH is designed to be straightforward and user-friendly, removing many of the barriers that can deter people from engaging with blockchain technology.

However, it's important to note that while ckETH provides these benefits, it's not a direct replacement for Bitcoin or ETH. Each has its own unique features and use cases, and the choice between them will depend on your specific needs and circumstances.

Sources:
- [A Data-Driven Exploration of ckETH](https://medium.com/dfinity/a-data-driven-exploration-of-ckETH-the-digital-twin-of-ether-on-the-internet-computer-36b762be72e7#b197)
- [The Purposes of the Internet Computer](https://medium.com/dfinity/announcing-internet-computer-mainnet-and-a-20-year-roadmap-790e56cbe04a#0289)
- [Internet Computer for Ethereum Developers](https://wiki.internetcomputer.org/wiki/The_Internet_Computer_for_Ethereum_Developers#A_very_brief_introduction_to_the_Internet_Computer)
- [Extend Bitcoin, Ethereum and other blockchains](https://wiki.internetcomputer.org/wiki/Extend_Bitcoin,_Ethereum_and_other_blockchains#firstHeading)
```

6. **Gasless Token Swaps**: Using ckETH, users can swap tokens for a few cents with 0 gas fees [[source]](https://internetcomputer.org/ethereum-integration).

7. **Web2 Integration**: Connect smart contracts to the world outside blockchain. Fetch real-time price data and more from Web2 [[source]](https://internetcomputer.org/ethereum-integration).

8. **Signing Ethereum Transactions**: ICP smart contracts are capable of offering the on-chain Ethereum full node API by using the HTTPS outcalls feature to securely query the Ethereum blockchain, and send transactions to it [[source]](https://internetcomputer.org/ethereum-integration).

In summary, ckETH provides a secure, cost-effective, and efficient way to interact with the Ethereum blockchain, making it a great choice for your application or service.

## What advantages does ckETH offer in terms of speed, fees, and security?

ckETH, or chain-key Ether, is an Internet Computer (ICP)-native token that represents Ether (ETH), the native token of Ethereum. It offers several advantages in terms of speed, fees, and security:

> 1. **Speed:** ckETH transactions on the Internet Computer have significantly lower latency compared to Ethereum. They can be transferred with 1-2 seconds finality, making them almost instant .
> 
> 2. **Fees:** Ethereum holders can interact with ckETH on the Internet Computer, which has significantly lower gas fees than on Ethereum. Users can swap tokens for a few cents with 0 gas fees .
> 
> 3. **Security:** ckETH bypasses attack vectors by directly calling Ethereum RPCs through HTTPS Outcalls, a feature that enables ICP canister smart contracts to call API requests directly. This minimizes the risk of data leaks, hacks, and breaches . Additionally, issuing and redeeming ckETH goes through Know Your Transaction (KYT) checks to ensure no tainted bitcoin enters the Internet Computer blockchain or is transferred out to tainted Bitcoin addresses .
> 
> 4. **Integration:** ckETH can be directly processed by smart contracts hosted on the Internet Computer, and can be seamlessly converted back to ETH anytime .
> 
> Sources:
> : [Internet Computer for Ethereum Developers](https://wiki.internetcomputer.org/wiki/The_Internet_Computer_for_Ethereum_Developers#mw-toc-heading)
> : [A data-driven exploration of ckETH](https://medium.com/dfinity/a-data-driven-exploration-of-ckETH-the-digital-twin-of-ether-on-the-internet-computer-36b762be72e7#b197)
> : [How to Acquire ckETH](https://medium.com/dfinity/how-to-acquire-ckETH-02d863c835fc#c324)
> : [Trustless multi-chain on ICP](https://internetcomputer.org/multichain)
> : [Ethereum &lt;&gt; ICP](https://internetcomputer.org/ethereum-integration)
> : [Multi-chain DeFi](https://internetcomputer.org/defi)
> : [Why Bitcoin needs smart contracts](https://medium.com/dfinity/why-bitcoin-needs-smart-contracts-5191fbec294a#acf6)

Reviewer: ```suggestion
1. **Speed:** ckETH transactions on ICP have significantly lower latency compared to Ethereum. They can be transferred with 1-2 seconds finality, making them almost instant .

2. **Fees:** Ethereum holders can interact with ckETH on ICP, which has significantly lower gas fees than on Ethereum. Users can swap tokens for a few cents with 0 gas fees .

3. **Security:** ckETH bypasses attack vectors by directly calling Ethereum RPCs through HTTPS outcalls, a feature that enables ICP canister smart contracts to call API requests directly. This minimizes the risk of data leaks, hacks, and breaches . Additionally, issuing and redeeming ckETH goes through Know Your Transaction (KYT) checks to ensure no tainted bitcoin enters the Internet Computer blockchain or is transferred out to tainted Bitcoin addresses .

4. **Integration:** ckETH can be directly processed by smart contracts hosted on ICP, and can be seamlessly converted back to ETH anytime .

Sources:
- [Internet Computer for Ethereum Developers](https://wiki.internetcomputer.org/wiki/The_Internet_Computer_for_Ethereum_Developers#mw-toc-heading)
- [A data-driven exploration of ckETH](https://medium.com/dfinity/a-data-driven-exploration-of-ckETH-the-digital-twin-of-ether-on-the-internet-computer-36b762be72e7#b197)
- [How to Acquire ckETH](https://medium.com/dfinity/how-to-acquire-ckETH-02d863c835fc#c324)
- [Trustless multi-chain on ICP](https://internetcomputer.org/multichain)
- [Ethereum &lt;&gt; ICP](https://internetcomputer.org/ethereum-integration)
- [Multi-chain DeFi](https://internetcomputer.org/defi)
- [Why Bitcoin needs smart contracts](https://medium.com/dfinity/why-bitcoin-needs-smart-contracts-5191fbec294a#acf6)
```

## How can I integrate ckETH into my existing applications or platforms?

To integrate ckETH into your existing applications or platforms, you can follow these steps:

1. **Obtain an ICP principal or wallet address.**
2. **Deposit your ETH**: Ethereum holders can deposit Ether to the *deposit* function in the ckETH helper contract on Ethereum and specify their ICP principal or wallet address where they want their minted ckETH to appear.
3. **Mint ckETH**: The ICP ckETH canister smart contract will then mint the same amount of ckETH to the indicated ICP principal or wallet address.
4. **Use ckETH**: The ICP principal or wallet address can now use ckETH natively on the Internet Computer.
5. **Convert back to ETH any time**: When the ICP principal wants to convert ckETH back to ETH, they can request the ckETH canister smart contract with the ETH amount and Ethereum address to which they want to send the ETH.

You can learn more about how ckETH works from this [DFINITY forum post](https://forum.dfinity.org/t/ckETH-a-canister-issued-ether-twin-token-on-the-ic/22819).

> Please note that the ckETH integration is part of a larger Ethereum integration on ICP. You can learn more about the Ethereum integration on ICP from this [website page](https://internetcomputer.org/ethereum-integration).

Reviewer: ```suggestion
Please note that the ckETH integration is part of a larger Ethereum integration on ICP. [Learn more about the Ethereum integration on ICP](https://internetcomputer.org/ethereum-integration).
```

Also, remember that the Internet Computer makes it possible to build almost any online service fully on-chain, in a full stack decentralization model. This is because canister smart contracts can hold up to 400GiB of memory each, and run in parallel with great efficiency. Moreover, they can directly serve interactive web-based user experiences to users by processing HTTP requests, thanks to ICP’s reverse-gas model (canister smart contracts pay for their own execution using “cycles” that they have been charged with). Now they can also be trustlessly combined with DeFi and other functionality Ethereum hosts in a World Computer paradigm. You can learn more about this from [here](https://internetcomputer.org/multichain).

## What are the technical requirements for developers to work with ckETH?

> WIP

Reviewer: ```suggestion
:::info
This answer is a work in progress. Please check back later for an updated response.
:::
```

The governance of the Internet Computer, and by extension ckETH, is decentralized. The evolution of the system depends on the governance mechanism that defines how a system can change. In the case of the Internet Computer, the NNS canisters define the governance mechanisms, including the parameters for staking, voting, and rewards. Users can stake ICP to participate in voting. The DFINITY foundation and the ICA are the parties with the highest voting power in the ecosystem, together holding less than 23% of the total voting power.

For more information on the governance and tokenomics of the Internet Computer, you can refer to the [Internet Computer Wiki](https://wiki.internetcomputer.org/wiki/NNS_Canisters) and [Tokenomics of a DAO](https://wiki.internetcomputer.org/wiki/Tokenomics_of_a_DAO).

: [Internet Computer Wiki - Decentralization](https://wiki.internetcomputer.org/wiki/Decentralization#firstHeading)

## Can you provide examples of current applications that utilize ckETH?

> 

Reviewer: ```suggestion
:::info
This answer is a work in progress. Please check back later for an updated response.
:::
```

# What is the current state of ckETH project?

> As of the latest updates from the provided extracts, ckETH is live on the Internet Computer mainnet. It has been added to the ICP dashboard and is also supported in the ICRC-1 wallet. However, with the current ckETH integration, you cannot withdraw to ETH directly from the ICRC-1 wallet. 
> 
> The DFINITY R&D teams and members of the ecosystem are working on enhancing the Ethereum with Internet Computer capabilities. The integration is being carried out in phases. Phase 0, which enables chain-key signing and HTTPS outcalls, is already complete. Phase 1 is being built to allow ETH API calls by accessing Web-2 based API providers through HTTPS outcalls. This phase also includes designing and building ckETH and ckERC-20 tokens. The final phase, Phase 2, will revolve around native ETH integration.
> 
> For more information on the current state of the ckETH project, you can join the forum discussion or check out the Ethereum integration page on the Internet Computer website.
> 
> : [Internet Computer Blog - ckETH: now live!](https://internetcomputer.org/blog/2023/12/06/news-and-updates/update#ckETH-now-live)
> : [Dfinity - Global R&D: May 2023 Edition - ETH Integration: Status update](https://medium.com/dfinity/global-r-d-may-2023-edition-d96859039ca6#a28d)
> : [Ethereum &lt;&gt; ICP - Multi-chain DeFi](https://internetcomputer.org/ethereum-integration)
> : [Ethereum &lt;&gt; ICP - Build your own Ethereummulti-chain solution - Tutorial](https://internetcomputer.org/ethereum-integration)

Reviewer: ```suggestion
ckETH is live on the Internet Computer mainnet. It has been added to the ICP dashboard and is also supported in the ICRC-1 wallet. However, with the current ckETH integration, you cannot withdraw to ETH directly from the ICRC-1 wallet. 

The integration is being carried out in phases. Phase 0, which enables chain-key signing and HTTPS outcalls, is already complete. Phase 1 is being built to allow ETH API calls by accessing Web-2 based API providers through HTTPS outcalls. This phase also includes designing and building ckETH and ckERC-20 tokens. The final phase, Phase 2, will revolve around native ETH integration.

For more information on the current state of the ckETH project, you can join the forum discussion or check out the Ethereum integration page on the ICP website.

- [Internet Computer Blog - ckETH: now live!](https://internetcomputer.org/blog/2023/12/06/news-and-updates/update#ckETH-now-live).
- [Dfinity - Global R&D: May 2023 Edition - ETH Integration: Status update](https://medium.com/dfinity/global-r-d-may-2023-edition-d96859039ca6#a28d).
- [Ethereum + ICP - Multi-chain DeFi](https://internetcomputer.org/ethereum-integration).
- [Ethereum + ICP - Build your own Ethereummulti-chain solution - Tutorial](https://internetcomputer.org/ethereum-integration).
```
